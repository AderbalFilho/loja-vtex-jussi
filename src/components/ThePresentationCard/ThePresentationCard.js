import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import { JButton, JCard } from '@ui';

import './ThePresentationCard.scss';

function ThePresentationCard({ classes, offer }) {
  const className = classNames('the-presentation-card', classes);
  const tag = offer.action ? 'button' : 'a';
  const rest = offer.action
    ? { onClick: offer.action }
    : {
        href: offer.link,
        rel: 'noreferrer noopener',
        target: '_blank',
      };

  return (
    <JCard size="small" shadow="black" classes={className}>
      <div className="the-presentation-card__img-container">
        <img
          src={offer.img}
          alt={offer.imgAlt}
          className="the-presentation-card__img"
        />
      </div>
      <JButton
        tag={tag}
        radius="circle"
        color="pink"
        border="pink"
        classes="the-presentation-card__button"
        {...rest}
      >
        {offer.buttonText}
      </JButton>
    </JCard>
  );
}

const requiredPropsCheck = (props, propName, componentName) => {
  if (!props.link && !props.action) {
    return new Error(
      `One of 'link' or 'action' is required by '${componentName}' component.`
    );
  }
};

ThePresentationCard.propTypes = {
  classes: PropTypes.string,
  offer: PropTypes.shape({
    img: PropTypes.string.isRequired,
    imgAlt: PropTypes.string.isRequired,
    buttonText: PropTypes.string.isRequired,
    link: requiredPropsCheck,
    action: requiredPropsCheck,
  }).isRequired,
};

export default ThePresentationCard;
