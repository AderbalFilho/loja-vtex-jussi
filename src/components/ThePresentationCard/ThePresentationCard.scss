.the-presentation-card {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  position: relative;
}

.the-presentation-card__img-container {
  align-items: center;
  display: flex;
  height: 200px;
  justify-content: center;
  padding: 10px 10px 0;
  width: 100%;
}

.the-presentation-card__img {
  max-height: 190px;
  max-width: 190px;
}

.the-presentation-card__button {
  bottom: 0;
  margin: 0 auto 16px;
  position: absolute;
}
