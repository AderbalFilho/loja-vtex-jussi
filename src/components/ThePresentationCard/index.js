import ThePresentationCard from './ThePresentationCard';

export default ThePresentationCard;
