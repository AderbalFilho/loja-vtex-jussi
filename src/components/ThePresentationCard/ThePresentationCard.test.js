import React from 'react';
import TestRenderer from 'react-test-renderer';

import ThePresentationCard from './ThePresentationCard';

describe('ThePresentationCard', () => {
  const offers = [
    {
      img: 'link to fridgeImg',
      imgAlt: 'Refrigerador',
      buttonText: 'Mais Detalhes',
      link: '#',
    },
    {
      img: 'link to foodMixerImg',
      imgAlt: 'Batedeira',
      buttonText: 'Comprar em 12x',
      action: jest.fn(),
    },
  ];

  it('matches snapshot with link', () => {
    const testRenderer = TestRenderer.create(
      <ThePresentationCard offer={offers[0]} />
    );
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });

  it('matches snapshot with action', () => {
    const testRenderer = TestRenderer.create(
      <ThePresentationCard offer={offers[1]} />
    );
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });
});
