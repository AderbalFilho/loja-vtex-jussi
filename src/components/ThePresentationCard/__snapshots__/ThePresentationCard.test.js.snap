// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThePresentationCard matches snapshot with action 1`] = `
<div
  className="j-card j-card--rad-normal j-card--shadow j-card--size-small the-presentation-card"
>
  <div
    className="the-presentation-card__img-container"
  >
    <img
      alt="Batedeira"
      className="the-presentation-card__img"
      src="link to foodMixerImg"
    />
  </div>
  <button
    className="j-button j-button--back-transparent j-button--border-pink j-button--pink j-button--rad-circle the-presentation-card__button"
    onClick={[MockFunction]}
  >
    Comprar em 12x
  </button>
</div>
`;

exports[`ThePresentationCard matches snapshot with link 1`] = `
<div
  className="j-card j-card--rad-normal j-card--shadow j-card--size-small the-presentation-card"
>
  <div
    className="the-presentation-card__img-container"
  >
    <img
      alt="Refrigerador"
      className="the-presentation-card__img"
      src="link to fridgeImg"
    />
  </div>
  <a
    className="j-button j-button--back-transparent j-button--border-pink j-button--pink j-button--rad-circle j-button--link the-presentation-card__button"
    href="#"
    rel="noreferrer noopener"
    target="_blank"
  >
    Mais Detalhes
  </a>
</div>
`;
